### PR TITLE
Fix dataresend

### PR DESCRIPTION
**Pull request by jrvollmer**

### CODE REVIEW

File: DataProcessor/DataUnpacker.cpp [hunk 1]
```diff
@@ -38,7 +38,7 @@ E bytesToGeneralData(QByteArray data, int startPos, int endPos, E typeZero)
 
 DataUnpacker::DataUnpacker(QObject *parent) : QObject(parent)
 {
-    FILE* fp = fopen("../sc1-driver-io/sc1-data-format/format.json", "r"); // NOTE: Windows: "rb"; non-Windows: "r"
+    FILE* fp = fopen(".../sc1-driver-io/sc1-data-format/format.json", "r"); // NOTE: Windows: "rb"; non-Windows: "r"
```
Comment (jrvollmer) [inline DataProcessor/DataUnpacker.cpp]:
```suggestion
    FILE* fp = fopen("../sc1-driver-io/sc1-data-format/format.json", "r"); // NOTE: Windows: "rb"; non-Windows: "r"
```

File: telemetrylib/TCP.cpp [hunk 2]
```diff
@@ -4,10 +4,13 @@
 //
 #include "DTI.h"
 
+
 class TCP : public DTI {
 public:
     void sendData(QByteArray bytes, long long timestamp) override {
         qDebug()<<"sending via TCP";
+        bytes.push_front("<bl>");
+        bytes.push_back("</bl>");
```
Comment (jrvollmer) [inline telemetrylib/TCP.cpp:13]:
```suggestion
        bytes.push_front("<bl>");
        bytes.push_back("</bl>");
```
Can you please add this to the SQL sending as well?

File: telemetrylib/TCP.cpp [hunk 3]
```diff
@@ -62,14 +68,38 @@ public slots:
...
+                emit connectionStatusChanged();
+            } else if(connected && !connection) {
+                sock.close();
+                connection = true;
+                emit connectionStatusChanged();
+            }
+            usleep(50000);
+            sock.abort();
+        }
+    }
```
Comment (jrvollmer) [inline telemetrylib/TCP.cpp]: I tested out Charles' changes with this as well as port 4005. Unfortunately, it doesn't look like it works. Could you please drop it from this PR for now?

**Comment by jrvollmer** [comment]
@MCLiii ^^^

File: telemetrylib/TCP.cpp (hunk 3 above)
Comment (MCLiii) [reply telemetrylib/TCP.cpp]: It does seem to be working with my testing on a multiple device network. Without this thread the class can't detect network a disconnect without tcp closing packet, with leads to data being lost during tcp timeout period.

Comment (jrvollmer) [reply telemetrylib/TCP.cpp]: UPDATE: Turns out I'm just tired and stupid. The IP address that it was trying to connect to didn't match mine. It was only a few digits off, so I didn't notice until a few minutes ago. I haven't tested it extensively or with a full implementation of the engineering data distributor with connection checking on that side, so there will probably be some additional minor tweaking to do. However, it looks like it works well.

File: telemetrylib/TCP.cpp [hunk 4]
```diff
@@ -62,14 +68,38 @@ public slots:
...
+    std::thread *t;
+    std::atomic<bool> isConnected=false;
+
+    /**
+     * creates a thread that pings the Chase car's static IP to check connection
+     */
+    void checkConnection() {
+        QTcpSocket sock;
+        while(!isConnected) {
+            sock.connectToHost("196.186.1.16", 22);
```
Comment (jrvollmer) [inline telemetrylib/TCP.cpp]:
```suggestion
            sock.connectToHost("192.168.1.16", 4005);
```

**Comment by jrvollmer** [comment]
See #26 for new PR

**Pull request closed without merging**